room/runtime/src/main/java/androidx/room/SQLiteCopyOpenHelper.java

```diff
@@ -217,4 +251,35 @@ private void copyDatabaseFile(File destinationFile) throws IOException {
                     + destinationFile.getAbsolutePath() + ").");
         }
     }
+
+    private SupportSQLiteOpenHelper createFrameworkOpenHelper() {
+        SupportSQLiteOpenHelper frameworkOpenHelper = null;
+        try {
+            FrameworkSQLiteOpenHelperFactory factory = new FrameworkSQLiteOpenHelperFactory();
+            Configuration configuration = Configuration.builder(mContext)
+                    .name(mDatabaseConfiguration.name)
+                    .callback(new Callback(mDatabaseVersion){
```

Comment:
Lets not open the DB with the 'expected final version' we need to open the pre-package DB with the same version it already has, to avoid going through the migration path which will end up updating the version and when Room opens the DB again it will think its already updated. To find out the version of the DB without opening use https://github.com/androidx/androidx/blob/androidx-master-dev/room/runtime/src/main/java/androidx/room/util/DBUtil.java#L139

Reply:
I appreciate those explanations. It helps a lot. Thanks!